docker/Dockerfile.iroh-gateway

```diff
@@ -0,0 +1,51 @@
+################################################################################
+## Builder
+################################################################################
+FROM rust:latest AS builder
+
+RUN update-ca-certificates
+
+# install latest protocol buffer compiler. Yes, it's really this irritating.
+# recent build URLs are missing "3" version prefix. version is actually "3.21.9"
+ENV RUST_BACKTRACE=1 \
+  PROTOC_VERSION=21.9 \
+  PROTOC_ZIP=protoc-21.9-linux-aarch_64.zip \
+  # PROTOC_ZIP=protoc-21.9-linux-x86_64.zip \
+  PROTOC=/usr/local/bin/protoc \
+  PROTOC_INCLUDE=/usr/local/include
+
+RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOC_VERSION/$PROTOC_ZIP \
+  && unzip -o $PROTOC_ZIP -d /usr/local bin/protoc \
+  && unzip -o $PROTOC_ZIP -d /usr/local 'include/*' \
+  && rm -f $PROTOC_ZIP \
+  && echo $($PROTOC --version)
+
+# has the side effect of updating the crates.io index & installing rust toolchain
+# called in a separate step for nicer caching. the command itself will fail,
+# b/c empty-library is not a dependency, so we override with an exit code 0
+RUN cargo install empty-library; exit 0
+
+WORKDIR /iroh
+
+COPY ../ .
+
+RUN cargo build --bin iroh-gateway --release
+
+################################################################################
+## Final image
+################################################################################
+FROM gcr.io/distroless/cc
+
+WORKDIR /iroh
+
+# Copy our build, changing owndership to distroless-provided "nonroot" user,
+# (65532:65532)
+COPY --from=builder --chown=65532:65532 /iroh/target/release/iroh-gateway ./
```

Comment:
I would love to validate the file descriptor limits get raised/set correctly.

Reply:
I believe these are set at runtime, no?

Reply:
Yes, but docker is weird. It sets up to max ~65k at runtime but if not root the "hard limit" specified on the system level might be low (usually 1024). Since it's a non-root user, we might hit that easily. The proper thing should be setting the FD limit on the image directly as the system level config. Then the runtime can self manage.

Reply:
NVM just confirmed, this should be handled by the host system running docker, either by configuring the docker runner directly or supplying `--ulimit=12321312` when running stuff. Either way think this is a solid point for docs.